src/FNAPlatform/SDL2_FNAPlatform.cs

```diff
@@ -2567,6 +2567,38 @@ public static TouchPanelCapabilities GetTouchCapabilities()
 			};
 		}
 
+		public static void UpdateTouchPanelState()
+		{
+			// Poll the touch device for all active fingers
+			long touchDevice = SDL.SDL_GetTouchDevice(0);
+			for (int i = 0; i < TouchPanel.MAX_TOUCHES; i += 1)
+			{
+				IntPtr fingerPtr = SDL.SDL_GetTouchFinger(touchDevice, i);
+				if (fingerPtr == IntPtr.Zero)
+				{
+					// No finger found at this index
+					TouchPanel.SetFinger(i, TouchPanel.NO_FINGER, Vector2.Zero);
+					continue;
+				}
+
+				// Get finger info
+				SDL.SDL_Finger finger = (SDL.SDL_Finger) Marshal.PtrToStructure(
```

Comment:
Any chance we can turn this into an `unsafe { SDL.SDL_Finger* finger; }`? Just to avoid the variable duplication and such.